dynatrace/configuration_v1/anomaly_detection_process_groups.py

```diff
@@ -0,0 +1,43 @@
+from typing import List, Optional, Dict, Any
+from requests import Response
+
+from dynatrace.environment_v2.schemas import ConfigurationMetadata
+from dynatrace.environment_v2.monitored_entities import EntityShortRepresentation
+from dynatrace.pagination import PaginatedList
+from dynatrace.dynatrace_object import DynatraceObject
+from dynatrace.http_client import HttpClient
+
+class AvailabilityMonitoringPG(DynatraceObject):
+    def _create_from_raw_data(self, raw_element: Dict[str, Any]):
+        self.method: str = raw_element.get("method")
+        self.minimum_threshold: int = raw_element.get("minimumThreshold") 
+
+class AnomalyDetectionPG(DynatraceObject):
+    def _create_from_raw_data(self, raw_element: Dict[str, Any]):
+        self.availability_monitoring: AvailabilityMonitoringPG = AvailabilityMonitoringPG(raw_element=raw_element.get("availabilityMonitoring"))
+
+class AnomalyDetectionPGService:
+    def __init__(self, http_client: HttpClient):
+        self.__http_client = http_client
+
+    def get_configuration(self, id: str):
+        """
+        Get the anomaly detection configuration for the specified process group.
+        """
+        return AnomalyDetectionPG(raw_element=self.__http_client.make_request(f"/api/config/v1/anomalyDetection/processGroups/{id}").json())
+
+    def update_configuration(self, id: str, method: str, minimum_threshold: Optional[int] = None):
+        """
+        Update the anomaly detection configuration for the specified process group.
+        """
+        if method == "MINIMUM_THRESHOLD":
```

Comment:
Every time we have an `Enum` like `method`, which can be `[ "MINIMUM_THRESHOLD", "OFF", "PROCESS_IMPACT" ]`

Let's make the parameter an Enum as well (instead of a str), it will make it much easier for the user to know what are the valid methods he can use, so create an enum called `Method`, examples here:


https://github.com/dlopes7/dt-api/blob/69bb84784da6cb7861b9d0eeeb226bcd8c0af143/dynatrace/configuration_v1/extensions.py#L174

https://github.com/dlopes7/dt-api/blob/69bb84784da6cb7861b9d0eeeb226bcd8c0af143/dynatrace/configuration_v1/extensions.py#L220

Reply:
Done. And used the recommendation to allow a user to either supply a string or the Enum for the method.